internal/artifactory_cache_test.go

```diff
@@ -0,0 +1,139 @@
+			if r.Method == "PUT" { // Check for upload attempts.
+				uploadAttempted = true
+				t.Error("Should not attempt upload when NO_CACHE_UPLOAD is set")
+			}
+			w.Write(testContent)
+		}
+		// Start the HTTP server and get the port it runs on.
+		port, server := test.HttpHandler(handler)
+		defer server.Close() // Ensure the server is stopped after the test.
+
+		// Create a lock file with the resource and cache information.
+		lockContent := fmt.Sprintf(`[[Resource]]
+            Urls = ['http://localhost:%d/test.txt']
+            Integrity = '%s'
+            CacheUri = 'http://localhost:%d/cache'`, port, expectedHash, port)
+
+		lockPath := test.TmpFile(t, lockContent)
+		lock, err := NewLock(lockPath, false)
+		require.NoError(t, err) // Fail the test if the lock cannot be created.
+
+		// Download the resource into a temporary directory.
+		tmpDir := test.TmpDir(t)
+		err = lock.Download(tmpDir, []string{}, []string{}, "", false)
+		assert.NoError(t, err) // Verify the download succeeded.
+
+		// Ensure no upload was attempted.
+		assert.False(t, uploadAttempted)
+	})
+}
+
+// TestDeleteWithArtifactoryCache verifies deleting a resource with caching enabled.
+func TestDeleteWithArtifactoryCache(t *testing.T) {
+	// Sub-test to verify successful deletion of a resource.
+	t.Run("SuccessfulDelete", func(t *testing.T) {
+		// Set the GRABIT_ARTIFACTORY_TOKEN environment variable.
+		t.Setenv("GRABIT_ARTIFACTORY_TOKEN", "test-token")
+
+		// Setup an HTTP handler to handle DELETE requests.
+		handler := func(w http.ResponseWriter, r *http.Request) {
+			if r.Method == "DELETE" { // Respond with OK for DELETE requests.
+				w.WriteHeader(http.StatusOK)
+			}
+		}
+		// Start the HTTP server and get the port it runs on.
+		port, server := test.HttpHandler(handler)
+		defer server.Close() // Ensure the server is stopped after the test.
+
+		// Set up URLs for the source file and cache.
+		sourceURL := fmt.Sprintf("http://localhost:%d/test.txt", port)
+		cacheURL := fmt.Sprintf("http://localhost:%d", port)
+
+		// Create a lock file with the resource and cache information.
+		lockContent := fmt.Sprintf(`[[Resource]]
+            Urls = ['%s']
+            Integrity = 'sha256-test'
+            CacheUri = '%s'`, sourceURL, cacheURL)
+
+		lockPath := test.TmpFile(t, lockContent)
+		lock, err := NewLock(lockPath, false)
+		require.NoError(t, err) // Fail the test if the lock cannot be created.
+
+		// Save the lock file before modifying it.
+		err = lock.Save()
+		require.NoError(t, err) // Fail the test if saving fails.
+
+		// Delete the resource from the lock file.
+		lock.DeleteResource(sourceURL)
+
+		// Save the lock file again after deletion.
+		err = lock.Save()
+		require.NoError(t, err) // Fail the test if saving fails.
+
+		// Reload the lock file and verify the resource is gone.
+		newLock, err := NewLock(lockPath, false)
+		require.NoError(t, err)
+		assert.Equal(t, 0, len(newLock.conf.Resource)) // Ensure no resources remain.
```

Comment:
Here you're just testing that the resource deleting works... but that's not at all what the new code does... so why are you testing this at all?

Reply:
I think the test cases haven't been updated to align with the new code yet. Do you have please any suggestions that might help with updating the test cases to better reflect the changes?